modules/index-service/src/main/java/org/opencastproject/index/service/impl/IndexServiceImpl.java

```diff
@@ -1740,15 +1740,151 @@ public Source getEventSource(Event event) {
   }
 
   private void updateMediaPackageMetadata(MediaPackage mp, MetadataList metadataList) {
+    String oldSeriesId = mp.getSeries();
     List<EventCatalogUIAdapter> catalogUIAdapters = getEventCatalogUIAdapters();
-    if (catalogUIAdapters.size() > 0) {
+    if (catalogUIAdapters != null) {
       for (EventCatalogUIAdapter catalogUIAdapter : catalogUIAdapters) {
         Opt<MetadataCollection> metadata = metadataList.getMetadataByAdapter(catalogUIAdapter);
         if (metadata.isSome() && metadata.get().isUpdated()) {
           catalogUIAdapter.storeFields(mp, metadata.get());
         }
       }
     }
+
+    // update series catalogs
+    if (!StringUtils.equals(oldSeriesId, mp.getSeries())) {
+      List<String> seriesDcTags = new ArrayList<>();
+      List<String> seriesAclTags = new ArrayList<>();
+      Map<String, List<String>> seriesExtDcTags = new HashMap<>();
+      if (StringUtils.isNotBlank(oldSeriesId)) {
+        // remove series dublincore from the media package
+        for (MediaPackageElement mpe : mp.getElementsByFlavor(MediaPackageElements.SERIES)) {
+          mp.remove(mpe);
+          for (String tag : mpe.getTags()) {
+            seriesDcTags.add(tag);
+          }
+        }
+        // remove series ACL from the media package
+        for (MediaPackageElement mpe : mp.getElementsByFlavor(MediaPackageElements.XACML_POLICY_SERIES)) {
+          mp.remove(mpe);
+          for (String tag : mpe.getTags()) {
+            seriesAclTags.add(tag);
+          }
+        }
+        // remove series extended metadata from the media package
+        try {
+          Opt<Map<String, byte[]>> oldSeriesElementsOpt = seriesService.getSeriesElements(oldSeriesId);
+          for (Map<String, byte[]> oldSeriesElements : oldSeriesElementsOpt) {
+            for (String oldSeriesElementType : oldSeriesElements.keySet()) {
+              for (MediaPackageElement mpe : mp
+                      .getElementsByFlavor(MediaPackageElementFlavor.flavor(oldSeriesElementType, "series"))) {
+                mp.remove(mpe);
+                String elementType = mpe.getFlavor().getType();
+                if (StringUtils.isNotBlank(elementType)) {
+                  for (String tag : mpe.getTags()) {
+                    if (!seriesExtDcTags.containsKey(elementType)) {
+                      List<String> tags = new ArrayList<>();
+                      tags.add(tag);
+                      seriesExtDcTags.put(elementType, tags);
+                    } else {
+                      seriesExtDcTags.get(elementType).add(tag);
+                    }
+                  }
+                }
+              }
+            }
+          }
+        } catch (SeriesException e) {
+          logger.info("Unable to retrieve series element types from series service for the series {}", oldSeriesId, e);
+        }
+      }
+
+      if (StringUtils.isNotBlank(mp.getSeries())) {
+        // add updated series dublincore to the media package
```

Comment:
It seems that the media package update will just put the updated catalogs to the workspace, but at some point of time, the workspace cleaner will delete those files.

How are the catalogs made persistent?

Reply:
That is true. The callers take always a snapshot of the updated media package as described by @KatrinIhler.